core/src/main/php/xp/codegen/dataset/xp6.php.xsl

```diff
@@ -0,0 +1,529 @@
+      <xsl:otherwise>
+        <func:result select="concat($p, my:ucfirst($tname))" />
+      </xsl:otherwise>
+    </xsl:choose>
+  </func:function>
+
+  <func:function name="my:separator">
+    <xsl:param name="database"/>
+    <xsl:param name="table"/>
+    <xsl:param name="dbtype"/>
+    <xsl:choose>
+      <xsl:when test="$dbtype = 'mysql'"><func:result select="concat($database, '.', $table)"/></xsl:when>
+      <xsl:when test="$dbtype = 'sybase'"><func:result select="concat($database, '..', $table)"/></xsl:when>
+      <xsl:otherwise><func:result select="$table"/></xsl:otherwise>
+    </xsl:choose>
+  </func:function>
+  
+  <func:function name="my:distinctRole">
+    <xsl:param name="nodeset" />
+    <xsl:choose>
+      <xsl:when test="count($nodeset[@role != $nodeset[1]/@role]) &lt; 1"><func:result select="$nodeset[1]" /></xsl:when>
+      <xsl:otherwise><func:result select="$nodeset[@role = $nodeset[1]/@role][1] | my:distinctRole($nodeset[@role != $nodeset[1]/@role])" /></xsl:otherwise>
+    </xsl:choose>
+  </func:function>
+  
+  <func:function name="my:referenced">
+    <xsl:param name="datasetnode" />
+    <func:result select="my:distinctRole(document($constraintfile)/document/database[@database = $datasetnode/table/@database]/table/constraint/reference[@table = $datasetnode/table/@name])" />
+  </func:function>
+  
+  <func:function name="my:referencing">
+    <xsl:param name="datasetnode" />
+    <func:result select="my:distinctRole($datasetnode/table/constraint/reference)" />
+  </func:function>
+  
+  <func:function name="my:sameKeys">
+    <xsl:param name="keySet1" />
+    <xsl:param name="keySet2" />
+    <xsl:choose>
+      <xsl:when test="count($keySet1) != count($keySet2)"><func:result select="false()" /></xsl:when>
+      <xsl:when test="count($keySet1) = 0"><func:result select="true()" /></xsl:when>
+      <xsl:when test="$keySet1[1]/text() != $keySet2[1]/text()"><func:result select="false()" /></xsl:when>
+      <xsl:otherwise><func:result select="my:sameKeys($keySet1[position() != 1], $keySet2[position() != 1])" /></xsl:otherwise>
+    </xsl:choose>
+  </func:function>
+
+  <func:function name="my:distinctIndex">
+    <xsl:param name="indexSet" />
+    <xsl:choose>
+      <xsl:when test="count( $indexSet[not(my:sameKeys(key, $indexSet[1]/key))] ) &lt; 1"><func:result select="$indexSet[1]" /></xsl:when>
+      <xsl:otherwise><func:result select="$indexSet[1] | my:distinctIndex( $indexSet[not(my:sameKeys(key, $indexSet[1]/key))] )" /></xsl:otherwise>
+    </xsl:choose>
+  </func:function>
+
+  <func:function name="my:namespaceName">
+    <func:result select="translate($package, '.', '\')"/>
+  </func:function>
+
+  <xsl:template match="/">
+
+    <xsl:text>&lt;?php namespace </xsl:text><xsl:value-of select="my:namespaceName()"/>
+    <xsl:text>;
+
+use lang\XPClass;
+use rdbms\DataSet;
+use rdbms\Peer;
+use rdbms\Criteria;
+use rdbms\FieldType;
+use util\HashmapIterator;&#10;</xsl:text>
+    <xsl:apply-templates/>
```

Comment:
You'll need to change this to `rdbms\CachedResults` in the future, see xp-framework/rdbms@ba144db